src/cets_join.erl

```diff
@@ -135,10 +140,51 @@ join2(Info, LocalPid, RemotePid, JoinOpts) ->
         lists:foreach(fun({Pid, Ref}) -> catch cets:unpause(Pid, Ref) end, Paused)
     end.
 
-send_dump(Pid, Pids, JoinRef, Dump, JoinOpts) ->
+-spec pause_servers(AllPids :: [pid(), ...]) -> Paused :: [{pid(), cets:pause_monitor()}].
+pause_servers(AllPids) ->
+    %% We should create a pause helper process on each node in the cluster.
+    %% It is to ensure that node that loosing a connection with cets_join coordinator
+    %% would not unpause one of the processes too soon
+    %% (because it could start sending remote ops to nodes which are still in the current joining procedure).
+    Paused = [{Pid, cets:pause(Pid)} || Pid <- AllPids],
+    OtherNodes = lists:delete(node(), lists:usort([node(Pid) || Pid <- AllPids])),
+    Results = erpc:multicall(
+        OtherNodes, ?MODULE, pause_on_remote_node, [self(), AllPids], timer:seconds(30)
```

Comment:
I don't get it - why do we make all nodes call pause on all nodes? It results in a quadratic number of pauses - each node would get paused N-1 times when N is the total number of nodes.

Reply:
kinda, would lead to 10k pauses if there are 100 nodes.
Pause on all nodes is a stronger way to do the current pause logic.
It is just the simplest solution to ensure that we start processing messages only when all connected to us nodes removed their pause claim.

Reply:
Basically, main issue is that if node disconnects for some reason, it does not disappears from all nodes atomically or even fast. So, there could be a short period of time when node looses connection with the node that executing cets_join:join, but still keeps a connection to other nodes to the cluster.